source/features/restore-file.tsx

```diff
@@ -79,32 +87,19 @@ async function handleRestoreFileClick(event: delegate.Event<MouseEvent, HTMLButt
 
 	filesRestored.add(menuItem);
 
-	menuItem.textContent = 'Restoring…';
-	event.preventDefault();
-	event.stopPropagation();
-
 	try {
 		const filePath = menuItem.closest<HTMLDivElement>('[data-path]')!.dataset.path!;
-		const file = await getFile(filePath);
-
-		if (!file) {
-			// The file was created by this PR.
-			// This code won’t be reached if `highlight-deleted-and-added-files-in-diffs` works.
-			showError(menuItem, 'Nothing to restore. Delete file instead');
-			return;
-		}
-
-		if (file.isTruncated) {
-			showError(menuItem, 'Restore failed: File too big');
-			return;
-		}
-
-		await commitFileContent(menuItem, file.text, filePath);
+		// Show toast while restoring
+		void showToast(async progress => restoreFile(progress, menuItem, filePath), {
```

Comment:
We need to await this
```suggestion
		await showToast(async progress => restoreFile(progress, menuItem, filePath), {
```